runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryStateInternals.java

```diff
@@ -72,6 +72,16 @@ public K getKey() {
     return key;
   }
 
+  @Nullable
+  public Instant getWatermarkHold() {
+    for (State state : this.inMemoryState.values()) {
+      if (state instanceof WatermarkHoldState) {
+        return ((WatermarkHoldState<?>) state).read();
```

Comment:
Replaced with TestInMemoryStateInternals which already has a function like this.